code/atoms/objects/tree.dm

```diff
@@ -0,0 +1,46 @@
+/obj/tree
+	name = "tree"
+	desc = "a woody tree"
+	icon = 'icons/32x64.dmi'
+	icon_state = "tree"
+	density = TRUE
+	layer = ABOVE_MOB_LAYER
+
+/obj/tree/destroy()
+	new /obj/item/wood(loc)
+	return ..()
+
+/obj/item/wood
+	name = "wood"
+	desc = "wood for crafting. slap rocks on it with left click to make pickaxe, or right click to make sword"
+	icon_state = "wood"
+
+/obj/item/wood/left_click(adjacent, params)
+	if (!adjacent)
+		return
+	if (!istype(usr, /mob/living/inventory))
+		return
+	var/mob/living/inventory/M = usr
+	if(make_tool(M.inventory, /obj/item/pickaxe))
+		return TRUE
```

Comment:
left_click and right_click both get the item as the third argument, you can just use that instead of having to fuck around with the inventory.